pulp_ansible/app/tasks/signature.py

```diff
@@ -72,6 +91,71 @@ def verify_signature_upload(data):
     return data
 
 
+def verify_sigstore_signature_upload(data):
+    """The task code for verifying Sigstore signature upload."""
+    collection = data["signed_collection"]
+    repository = data.get("repository")
+    sigstore_signing_service = repository and repository.sigstore_signing_service
```

Comment:
This is the point, where when you look at the existing verify code, there is no notion of a signing service at all. An installation that is not configured to sign anything still needs to be able to verify.

Also can we have the one existing verify task determine the type of signature and handle accordingly?

Reply:
> An installation that is not configured to sign anything still needs to be able to verify.

I agree this would be a better approach. Repositories could be configured with a separate verifying service, the only thing is that users should be aware that some of the parameters passed to the signing service used to sign collections (or to the Sigstore client used if signing was done locally) should be taken into account when creating the verifying service (for example, they should both use the same URLs for the Rekor and TUF instances and the identity and OIDC issuers used to sign should be the ones checked in the certificate by the verifying service).

> Also can we have the one existing verify task determine the type of signature and handle accordingly?

Yes, we could write both verification tasks in the same method and use a parameter to differentiate between which type of verification should be used.